pkg/webhook/mutating/node/allocatable_mutator.go

```diff
@@ -41,154 +41,109 @@ func NewWebhookNodeAllocatableMutator() *WebhookNodeAllocatableMutator {
-			} else {
-				newAllocatable = &quantity
-				klog.V(6).Infof("node %s cpu allocatable by annotation: %v", node.Name, newAllocatable.String())
-			}
-		}
+	return nil
+}
 
-		// calculate allocatable and capacity by overcommit ratio
-		if newAllocatable == nil || newCapacity == nil {
-			CPUOvercommitRatio, err := cpuOvercommitRatioValidate(node.Annotations)
-			if err != nil {
-				klog.Errorf("node %s %s validate fail, value: %s, err: %v", node.Name, consts.NodeAnnotationCPUOvercommitRatioKey, CPUOvercommitRatioValue, err)
-			} else {
-				if CPUOvercommitRatio > 1.0 {
-					allocatable := node.Status.Allocatable.Cpu()
-					capacity := node.Status.Capacity.Cpu()
-					allocatableByOvercommit := native.MultiplyResourceQuantity(core.ResourceCPU, *allocatable, CPUOvercommitRatio)
-					capacityByOvercommit := native.MultiplyResourceQuantity(core.ResourceCPU, *capacity, CPUOvercommitRatio)
-					newAllocatable = &allocatableByOvercommit
-					newCapacity = &capacityByOvercommit
+func updateNodeAnnotations(annotations map[string]string, capacity, allocatable core.ResourceList) map[string]string {
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
 
-					klog.V(6).Infof(
-						"node %s %s capacity: %v, allocatable: %v, newCapacity: %v, newAllocatable: %v",
-						node.Name, core.ResourceCPU,
-						capacity.String(), newCapacity.String(),
-						allocatable.String(), newAllocatable.String())
-				}
-			}
-		}
+	annotations[consts.NodeAnnotationOriginalCapacityCPUKey] = capacity.Cpu().String()
+	annotations[consts.NodeAnnotationOriginalCapacityMemoryKey] = capacity.Memory().String()
+	annotations[consts.NodeAnnotationOriginalAllocatableCPUKey] = allocatable.Cpu().String()
+	annotations[consts.NodeAnnotationOriginalAllocatableMemoryKey] = allocatable.Memory().String()
 
-		if newAllocatable != nil && newCapacity != nil {
-			node.Status.Allocatable[core.ResourceCPU] = *newAllocatable
-			node.Status.Capacity[core.ResourceCPU] = *newCapacity
-		}
-	}
+	return annotations
+}
 
-	memoryOvercommitRatioValue, ok := node.Annotations[consts.NodeAnnotationMemoryOvercommitRatioKey]
+// processResource Update the resource information of the node according to the specified key-value pair
+func processResource(node *core.Node, rn core.ResourceName, ratioKey, capacityKey, allocatableKey string) error {
+	ratioValue, ok := node.Annotations[ratioKey]
 	if ok {
 		var newAllocatable, newCapacity *resource.Quantity
 		// get overcommit allocatable and capacity from annotation first
-		overcommitCapacity, ok := node.Annotations[consts.NodeAnnotationOvercommitCapacityMemoryKey]
+		capacity, ok := node.Annotations[capacityKey]
 		if ok {
-			quantity, err := resource.ParseQuantity(overcommitCapacity)
+			quantity, err := resource.ParseQuantity(capacity)
 			if err != nil {
-				klog.Error(err)
-			} else {
-				newCapacity = &quantity
-				klog.V(6).Infof("node %s mem capacity by annotation: %v", node.Name, newCapacity.String())
+				return err
```

Comment:
Would it better that just log error here? newAllocatable and newCapacity will be recalculated by overcommitRatio as a degradation.

Reply:
that makes sense. thank for reminding! done